server/gorm/deletions.go

```diff
@@ -71,6 +71,7 @@ func (c *Client) DeleteChildrenOfProject(ctx context.Context, project names.Proj
 // DeleteChildrenOfApi deletes all the children of a api.
 func (c *Client) DeleteChildrenOfApi(ctx context.Context, api names.Api) error {
 	for _, entityName := range []string{
+		storage.ArtifactEntityName,
```

Comment:
aside: I don't think we have a way (in testing) to verify that the blobs associated with specs and artifacts are actually getting deleted - this is less urgent than testing the API but worth noting

Reply:
I think if we want to test this we can do the following.
1. Create an API and child artifact. 
2. Delete the API.
3. Recreate the API. 
4. Read the artifact again.
5. Assert it doesn't exist.

I didn't include this test because it seems like a bug that's at low risk of regression, and has quite low cost even if we did reintroduce the bug.

Reply:
That mostly gets it - since artifacts (like specs) store contents in blobs, we might want to also verify that the blob was deleted. Blobs are implementation details and not exposed by the API, so this would be like looking for a memory leak.

Reply:
Functionally it should be invisible unless it causes issues recreating the artifact with the leftover blob still existing. If we tested directly from the gorm layer this can be easily verified though.